go/ql/lib/semmle/go/Types.qll

```diff
@@ -489,8 +485,7 @@ class StructType extends @structtype, CompositeType {
    */
   private predicate hasEmbeddedField(Type tp, int depth) {
     exists(Field f | this.hasFieldCand(_, f, depth, true) |
-      tp = f.getType() or
-      tp = f.getType().(PointerType).getBaseType()
+      tp = lookThroughPointerType(f.getType())
```

Comment:
Note this one may change behaviour, since `tp` would have had multiple values before (unlike all of the others, where the type of `tp` is disjoint with `PointerType`)

Reply:
Oops, that's a mistake. I skipped some other cases like that. I'll revert this one.